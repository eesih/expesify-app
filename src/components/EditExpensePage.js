import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, startRemoveExpense } from './../actions/expences';

export class EditExpensePage extends React.Component {

    onSubmitEditExpense = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }    
    onSubmitRemoveExpense = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <ExpenseForm 
                    expense={this.props.expense}
                    onSubmit={this.onSubmitEditExpense}
                />
                <button onClick={this.onSubmitRemoveExpense}>Remove</button>
            </div>
        )
    }
}

const mapStateToProps=(state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
    editExpense: (id, expense) => dispatch(editExpense(id, expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);