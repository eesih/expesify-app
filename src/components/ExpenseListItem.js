import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListitem = ({ description, id, amount, createdAt}) => (

    <div>
        <Link to={`/edit/${id}`}>{description}</Link>
        <p>{amount}</p>
        <p>{createdAt}</p>
    </div>
);

export default ExpenseListitem;