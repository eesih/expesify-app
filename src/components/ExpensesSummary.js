import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from './../selectors/expenses'; 
import getExpensesTotal from './../selectors/expenses-total';

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount == 1 ? 'expense' : 'expenses';
    return (
        <div> 
            <h1>{`Viewing ${expenseCount} ${expenseWord} totalling ${expensesTotal}`}</h1>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: numeral(getExpensesTotal(visibleExpenses) / 100).format('$0,0.00') 
    };
};

export default connect(mapStateToProps)(ExpensesSummary);