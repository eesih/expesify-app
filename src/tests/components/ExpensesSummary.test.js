import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={'$1,464.00'}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={'$3,421.00'}/>);
    expect(wrapper).toMatchSnapshot();
});