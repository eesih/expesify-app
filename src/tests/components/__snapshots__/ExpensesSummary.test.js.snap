// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div>
  <h1>
    Viewing 1 expense totalling $1,464.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 2 expenses totalling $3,421.00
  </h1>
</div>
`;
